crates/orchestrator/src/database/mod.rs

```diff
@@ -36,6 +36,8 @@ pub trait Database: Send + Sync {
 
     async fn update_metadata(&self, job: &JobItem, metadata: HashMap<String, String>) -> Result<()>;
     async fn get_latest_job_by_type_and_internal_id(&self, job_type: JobType) -> Result<Option<JobItem>>;
+
+    async fn get_successful_snos_jobs_without_proving(&self) -> Result<Vec<JobItem>>;
```

Comment:
I think the database trait shouldn't have functions with the name `snos` or `proving` as that seems to leak the internal types. We should generalise the function over all types and take the type as an input. Function can look like

```rust
fn get_jobs_without_successor(job_a_type, job_a_status, job_b_type, job_b_status)
```

wdyt?

Reply:
makes sense